src/node_realm.h

```diff
@@ -0,0 +1,79 @@
+#ifndef SRC_NODE_REALM_H_
+#define SRC_NODE_REALM_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <v8.h>
+#include "env_properties.h"
+#include "memory_tracker.h"
+#include "node_snapshotable.h"
+
+namespace node {
+
```

Comment:
Can I ask you to add some code comments here explaining Realms a bit more, what they are, what it does, etc. Just helpful for others coming along later

Reply:
Yeah, definitely. I'll sum up an abstract of the interface and describes its designs.